src/main/java/com/cloudbees/jenkins/support/api/FileContent.java

```diff
@@ -97,10 +145,35 @@ protected InputStream getInputStream() throws IOException {
     }
 
     @Override
-    public long getTime()  throws IOException {
+    public long getTime() throws IOException {
         return file.lastModified();
     }
 
+    // Check if the file is binary or not
```

Comment:
This is already handled in another class: https://github.com/jenkinsci/support-core-plugin/blob/master/src/main/java/com/cloudbees/jenkins/support/util/OutputStreamSelector.java

This logic should be added there to avoid double-detection.

Reply:
There is no double-detection since `FileContent` now extends `PrefilteredContent` and `shouldBeFiltered` will return `false`. That makes that the OutputStream will be `unfilteredOut`: https://github.com/jenkinsci/support-core-plugin/blob/128417c63460002be5bfa7411a3056fe6ab82478/src/main/java/com/cloudbees/jenkins/support/SupportPlugin.java#L304 
`unfilteredOut` is directly the binary stream: https://github.com/jenkinsci/support-core-plugin/blob/128417c63460002be5bfa7411a3056fe6ab82478/src/main/java/com/cloudbees/jenkins/support/SupportPlugin.java#L292
My intention here is to detect if the file to filter is binary or text, so in case it's binary it will not be filtered, and in case it's a text file it will be read and filtered line by line.
After having a look into `OutputStreamSelector`, I think how it's checking if the stream is binary or not is better than the one proposed here, so I will take it out from the class and put it into a utility class, so that check can be invoked from both places.